pkg/hub/controllers/cluster/reconciler.go

```diff
@@ -144,14 +143,21 @@ func (r *Reconciler) updateClusterInfo(ctx context.Context, cr *hubv1alpha1.Clus
 		return err
 	}
 	log.Info("got clusterinfo", "ci", clusterInfo)
-	if clusterInfo.ClusterProperty.GeoLocation.CloudProvider == GCP || clusterInfo.ClusterProperty.GeoLocation.CloudProvider == AWS || clusterInfo.ClusterProperty.GeoLocation.CloudProvider == AZURE {
-		cr.Spec.ClusterProperty.GeoLocation.CloudRegion = clusterInfo.ClusterProperty.GeoLocation.CloudRegion
-		cr.Spec.ClusterProperty.GeoLocation.CloudProvider = clusterInfo.ClusterProperty.GeoLocation.CloudProvider
-		if err := r.Update(ctx, cr); err != nil {
-			log.Error(err, "Error updating ClusterProperty on hub cluster")
-			return err
+	if clusterInfo.ClusterProperty.GeoLocation.CloudProvider == GCP || clusterInfo.ClusterProperty.GeoLocation.CloudProvider == AWS ||
+		clusterInfo.ClusterProperty.GeoLocation.CloudProvider == AZURE {
+		if clusterInfo.ClusterProperty.GeoLocation.CloudRegion != clusterInfo.ClusterProperty.GeoLocation.CloudRegion ||
```

Comment:
is this condition correct , clusterInfo.ClusterProperty.GeoLocation.CloudRegion != clusterInfo.ClusterProperty.GeoLocation.CloudRegion ? maybe a typo ?